controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/AccessControlEntry.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth.handler.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import io.pravega.auth.AuthHandler;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+
+import java.util.regex.Pattern;
+
+/**
+ * An entry of an {@link AccessControlList}.
+ */
+class AccessControlEntry {
+    private static final Pattern PATTERN_STAR_NOT_PRECEDED_BY_DOT = Pattern.compile("(?!=.)\\*");
+
+    @Getter(AccessLevel.PACKAGE)
+    private final String resourcePattern;
+
+    @Getter(AccessLevel.PACKAGE)
+    private final AuthHandler.Permissions permissions;
+
+    @VisibleForTesting
+    AccessControlEntry(@NonNull String aceResource, @NonNull AuthHandler.Permissions permissions) {
+        this(aceResource, permissions, false);
+    }
+
+    @VisibleForTesting
+    AccessControlEntry(@NonNull String aceResource, @NonNull AuthHandler.Permissions permissions, boolean isLegacyFormat) {
+        // Replaces any `*` with `.*`, if it's not already preceded by `.`, for regex processing.
+        // So, `pravega:://*` becomes `pravega:://.*` and `pravega:://scope:*` becomes `pravega:://scope:.*`
+        this.resourcePattern = isLegacyFormat ?  aceResource :
```

Comment:
so legacy pattern here does not have the "domain". and we do not have .* there. right?
so in the new pattern, why do we allow for `pravega:://scope:*` at all if this is the new pattern and we want ".*"

Reply:
I added support for `*` in lieu of the regex `.*` for usability. Admins/operators may find it easier to remember to use wildcards without the preceding dot. Both'd work exactly the same way as we replace `*` with `.*`  if it is not preceded by the dot.